src/Web5.js

```diff
@@ -0,0 +1,223 @@
+import nacl from 'tweetnacl';
+import * as DwnSDK from '@tbd54566975/dwn-sdk-js';
+
+import { DIDRegistry } from './did/DIDRegistry.js';
+import { Protocols } from './interface/Protocols.js';
+import { Records } from './interface/Records.js';
+import { LocalStorage } from './storage/LocalStorage.js';
+import { AppTransport } from './transport/AppTransport.js';
+import { HTTPTransport } from './transport/HTTPTransport.js';
+import { decodePin, parseJSON, triggerProtocolHandler } from './utils.js';
+
+class Web5 extends EventTarget {
```

Comment:
can you elaborate what you mean by this?
> using an event emitter where its needed

Reply:
Using an event emitter for connect() just like any app that you are using events for.  In either client/app or agent/server...

Not exactly the same, but first one that came to mind:
https://github.com/TBD54566975/dwn-server/blob/2c1474ac5fd337e46b2aa0cb7aa58d27fa943139/src/client/ws.js

Reply:
in order to use an `EventEmitter` we'd need to expose that it for clients to add event listeners to it

a benefit of `EventTarget` is that it's baked right into the object you're calling the method on

rather than having to do something like
```
web5.connectEvents.on('connect', ...)
web5.connect();
```
you can instead do
```
web5.addEventListener('connect', ...)
web5.connect();
```

Reply:
also unlike `EventEmitter` we get `EventTarget` for free cause it's a builtin in _all_ environments